GraphView/graphview.py

```diff
@@ -1552,145 +1540,129 @@ def set_current_list_desc(self, active_person):
 
-                # Add every child recursively
+                # add every child recursively
                 for child_ref in family.get_child_ref_list():
                     self.add_descendant(
-                        self.database.get_person_from_handle(child_ref.ref),
-                        num_generations - 1,
-                        person_handles)
+                           self.database.get_person_from_handle(child_ref.ref),
+                           num_generations - 1,
+                           person_handles)
 
-                # Add spouses
-                if person.handle == family.get_father_handle():
-                    spouse_handle = family.get_mother_handle()
-                else:
-                    spouse_handle = family.get_father_handle()
-
-                # add spouse itself
-                if spouse_handle and spouse_handle not in person_handles:
-                   person_handles.append(spouse_handle)
-
-                # add all his(her) spouses recursively
-                sp_person = None
-                if spouse_handle:
-                    sp_person = self.database.get_person_from_handle(spouse_handle)
-                if sp_person:
-                  for sp_family_handle in sp_person.get_family_handle_list():
-                     sp_family = self.database.get_family_from_handle(sp_family_handle)
-                     if sp_family.get_mother_handle() and sp_family.get_mother_handle() not in person_handles:
-                        self.add_descendant(
-                          self.database.get_person_from_handle(sp_family.get_mother_handle()),
-                          1,
-                          person_handles)
-                     if sp_family.get_father_handle() and sp_family.get_father_handle() not in person_handles:
-                        self.add_descendant(
-                          self.database.get_person_from_handle(sp_family.get_father_handle()),
-                          1,
-                          person_handles)
+                self.add_spouses(person, family, person_handles)
+
+    def add_spouses(self, person, family, person_handles):
+        """
+        Add spouses to the list.
+        """
+        # get spouse
+        if person.handle == family.get_father_handle():
+            spouse_handle = family.get_mother_handle()
+        else:
+            spouse_handle = family.get_father_handle()
+
+        # add spouse itself
+        if spouse_handle and spouse_handle not in person_handles:
+            person_handles.append(spouse_handle)
+
+        # add all his(her) spouses recursively
+        if spouse_handle:
+            sp_person = self.database.get_person_from_handle(spouse_handle)
+
+        if sp_person:
+            for family_handle in sp_person.get_family_handle_list():
+                sp_family = self.database.get_family_from_handle(family_handle)
+
+                m_handle = sp_family.get_mother_handle()
```

Comment:
read mother_handle and father_handle just once (instead of 3 times).
may little increase speed.